### PR TITLE
NormalizeL2 spec revision

### PR DESCRIPTION
**Pull request by mitruska**
### Details:
NormalizeL2 spec revision:
- [x] Update detailed description (operation formula stays unchanged)
- [x] Unify eps applying order in attributes description (apply *eps* before sqrt)
- [x] Add Types section
- [x] Update examples

### Tickets:
 - 40725 (45806)

### CODE REVIEW

File: docs/ops/normalization/NormalizeL2_1.md [hunk 1]
```diff
@@ -4,7 +4,22 @@
 
 **Category**: *Normalization*
 
-**Short description**: *NormalizeL2* operation performs L2 normalization of the 1st input tensor in slices specified by the 2nd input.
+**Short description**: *NormalizeL2* operation performs L2 normalization of the first input tensor in slices specified by the second input.
```
Comment (ggalieroc) [inline docs/ops/normalization/NormalizeL2_1.md:7]: Personally I like to indicate names of inputs if there is more than 1. In addition, I think we can omit tensor here as in inputs/outputs sections specify that inputs and outputs are tensors. Similarly as in [here](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/reduction/ReduceL2_4.md).

Something like:
performs L2 normalization on a given input \`data\` along dimensions specified by \`axes\` input

File: docs/ops/normalization/NormalizeL2_1.md [hunk 2]
```diff
@@ -4,7 +4,22 @@
...
+
+**Detailed Description**
+
+Each element in the output is the result of division of corresponding element from the `data` input tensor by the result of L2 reduction along dimensions specified by the `axes` input:
```
Comment (ggalieroc) [inline docs/ops/normalization/NormalizeL2_1.md:11]: the result of division of corresponding element from the \`data\` input tensor by the result of L2 reduction ---> the result of dividing the corresponding element of \`data\` input by the result of L2 reduction

File: docs/ops/normalization/NormalizeL2_1.md [hunk 3]
```diff
@@ -4,7 +4,22 @@
...
+
+    output[i0, i1, ..., iN] = x[i0, i1, ..., iN] / sqrt(eps_mode(sum[j0,..., jN](x[j0, ..., jN]**2), eps))
+
+Where indices `i0, ..., iN` run through all valid indices for the `data` input and summation `sum[j0, ..., jN]` have `jk = ik` for those dimensions `k` that are not in the set of indices specified by the `axes` input of the operation.
+`eps_mode` selects how the reduction value and `eps` are combined. It can be `max` or `add` depending on `eps_mode` attribute value.
+
+Particular cases:
+
+1. If `axes` is an empty list, then each input element is divided by itself resulting value `1` for all non-zero elements.
+2. If `axes` contains all dimensions of input `data`, a single L2 reduction value is calculated for entire input tensor and each input element is divided by that value.
```
Comment (ggalieroc) [inline docs/ops/normalization/NormalizeL2_1.md:21]: for entire input tensor ---> for the entire input tensor

File: docs/ops/normalization/NormalizeL2_1.md [hunk 4]
```diff
@@ -4,7 +4,22 @@
...
+
 
 **Attributes**
```
Comment (ggalieroc) [inline docs/ops/normalization/NormalizeL2_1.md:25]: I think you can remove this line in \*eps\* attribute: For example, \*eps\* equal to 0.001 means that 0.001 is used if all the values in normalization are equal to zero.

Additionally maybe would be nice to rewrite the description and avoid using added/maximized and to/with

File: docs/ops/normalization/NormalizeL2_1.md [hunk 5]
```diff
@@ -19,32 +34,52 @@
...
+* **2**: `axes` - Axis indices of `data` input tensor, along which L2 reduction is calculated. A scalar or 1D tensor of unique elements and type *T_IND*. The range of elements is `[-r, r-1]`, where `r` is the rank of `data` input tensor. **Required.**
 
 **Outputs**
 
-* **1**: Tensor of the same shape and type as the `data` input and normalized slices defined by `axes` input.
-
-**Detailed Description**
-
-Each element in the output is the result of division of corresponding element from the `data` input tensor by the result of L2 reduction along dimensions specified by the `axes` input:
+* **1**: Normalized tensor of the same shape and type as the `data` input.
```
Comment (ggalieroc) [inline docs/ops/normalization/NormalizeL2_1.md:50]: Suggestion: As other normalization operations. As we define formula/function above, mention it. Types usually go before shape in description.

\* \*\*1\*\*: The result of \*NormalizeL2\* function applied to \`data\` input tensor. Normalized tensor of the same type and shape as the `data` input.

File: docs/ops/normalization/NormalizeL2_1.md [hunk 6]
```diff
@@ -19,32 +34,52 @@
...
 
-    output[i0, i1, ..., iN] = x[i0, i1, ..., iN] / sqrt(eps_mode(sum[j0,..., jN](x[j0, ..., jN]**2), eps))
+**Types**
 
-Where indices `i0, ..., iN` run through all valid indices for the 1st input and summation `sum[j0, ..., jN]` have `jk = ik` for those dimensions `k` that are not in the set of indices specified by the `axes` input of the operation. One of the corner cases is when `axes` is an empty list, then we divide each input element by itself resulting value 1 for all non-zero elements. Another corner case is where `axes` input contains all dimensions from `data` tensor, which means that a single L2 reduction value is calculated for entire input tensor and each input element is divided by that value.
+* *T*: arbitrary supported floating-point type.
+* *T_IND*: any supported integer type.
 
-`eps_mode` selects how the reduction value and `eps` are combined. It can be `max` or `add` depending on `eps_mode` attribute value.
+**Examples**
```
Comment (ggalieroc) [inline docs/ops/normalization/NormalizeL2_1.md:58]: Suggestion: Maybe you could add headers to examples to spot the difference between them.
Like in [here](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/normalization/BatchNormInference_5.md)

File: docs/ops/normalization/NormalizeL2_1.md (hunk 1 above)
Comment (mitruska) [reply docs/ops/normalization/NormalizeL2_1.md]: Done

File: docs/ops/normalization/NormalizeL2_1.md (hunk 4 above)
Comment (mitruska) [reply docs/ops/normalization/NormalizeL2_1.md:25]: Done

File: docs/ops/normalization/NormalizeL2_1.md (hunk 2 above)
Comment (mitruska) [reply docs/ops/normalization/NormalizeL2_1.md]: Done

File: docs/ops/normalization/NormalizeL2_1.md (hunk 3 above)
Comment (mitruska) [reply docs/ops/normalization/NormalizeL2_1.md]: Done

File: docs/ops/normalization/NormalizeL2_1.md (hunk 4 above)
Comment (mitruska) [reply docs/ops/normalization/NormalizeL2_1.md:25]: Done

File: docs/ops/normalization/NormalizeL2_1.md (hunk 5 above)
Comment (mitruska) [reply docs/ops/normalization/NormalizeL2_1.md]: Thank you, done.

File: docs/ops/normalization/NormalizeL2_1.md (hunk 6 above)
Comment (mitruska) [reply docs/ops/normalization/NormalizeL2_1.md:58]: Done

**Review by ggalieroc — APPROVED** [review]

**Comment by ilyachur** [comment]
@rkazants Could you have a look?

File: docs/ops/normalization/NormalizeL2_1.md [hunk 7]
```diff
@@ -4,47 +4,86 @@
...
-**Short description**: *NormalizeL2* operation performs L2 normalization of the 1st input tensor in slices specified by the 2nd input.
+**Short description**: *NormalizeL2* operation performs L2 normalization on a given input `data` along dimensions specified by `axes` input.
+
+**Detailed Description**
+
+Each element in the output is the result of dividing the corresponding element of `data` input by the result of L2 reduction along dimensions specified by the `axes` input:
+
+    output[i0, i1, ..., iN] = x[i0, i1, ..., iN] / sqrt(eps_mode(sum[j0,..., jN](x[j0, ..., jN]**2), eps))
+
+Where indices `i0, ..., iN` run through all valid indices for the `data` input and summation `sum[j0, ..., jN]` have `jk = ik` for those dimensions `k` that are not in the set of indices specified by the `axes` input of the operation.
```
Comment (rkazants) [inline docs/ops/normalization/NormalizeL2_1.md:15]: summation `sum[j0, ..., jN]` have `jk = ik` -> summation `sum[j0, ..., jN]` has `jk = ik`

File: docs/ops/normalization/NormalizeL2_1.md [hunk 8]
```diff
@@ -4,47 +4,86 @@
...
   * **Required**: *yes*
 
 * *eps_mode*
 
-  * **Description**: Specifies how *eps* is combined with L2 value calculated before division.
-  * **Range of values**: `add`, `max`
+  * **Description**: Specifies how *eps* is combined with the sum of squares to avoid division by zero.
+  * **Range of values**: `add` or `max`
   * **Type**: `string`
   * **Default value**: None
```
Comment (rkazants) [inline docs/ops/normalization/NormalizeL2_1.md:39]: Not needed to mention `default value` here. Please remove it.

File: docs/ops/normalization/NormalizeL2_1.md [hunk 9]
```diff
@@ -4,47 +4,86 @@
...
 
 **Attributes**
 
 * *eps*
 
-  * **Description**: *eps* is the number to be added/maximized to/with the variance to avoid division by zero when normalizing the value. For example, *eps* equal to 0.001 means that 0.001 is used if all the values in normalization are equal to zero.
+  * **Description**: *eps* is the number applied by *eps_mode* function to the sum of squares to avoid division by zero when normalizing the value.
   * **Range of values**: a positive floating-point number
   * **Type**: `float`
   * **Default value**: None
```
Comment (rkazants) [inline docs/ops/normalization/NormalizeL2_1.md:31]: Not needed to mention default value here. Please remove it.

File: docs/ops/normalization/NormalizeL2_1.md (hunk 7 above)
Comment (mitruska) [reply docs/ops/normalization/NormalizeL2_1.md]: Thanks, done

File: docs/ops/normalization/NormalizeL2_1.md (hunk 9 above)
Comment (mitruska) [reply docs/ops/normalization/NormalizeL2_1.md]: Thanks, done

File: docs/ops/normalization/NormalizeL2_1.md (hunk 8 above)
Comment (mitruska) [reply docs/ops/normalization/NormalizeL2_1.md]: Thanks, done

**Review by rkazants — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Pull request merged**